### PR TITLE
Ensure nginx worker processes respects docker limits

### PR DESCRIPTION
**Pull request by felipecrs**
Refs https://github.com/blakeblackshear/frigate/issues/10969#issuecomment-2143025268

### CODE REVIEW

**Comment by felipecrs** [comment]
I tested by starting the devcontainer, then checking the contents of `/usr/local/nginx/conf/nginx.conf`.

You can limit the cpus in `docker-compose.yaml` with:

```yaml
services:
  devcontainer:
    deploy:
      resources:
        limits:
          cpus: '2'
```

File: docker/main/Dockerfile [hunk 1]
```diff
@@ -30,6 +30,9 @@ RUN --mount=type=tmpfs,target=/tmp --mount=type=tmpfs,target=/var/cache/apt \
     --mount=type=cache,target=/root/.ccache \
     /deps/build_nginx.sh
 
+# https://github.com/felipecrs/cgroup-scripts/commits/master/get_cpus.sh
+ADD --link --chmod=755 "https://github.com/felipecrs/cgroup-scripts/raw/178c3ed0cb55465af6652fbd2b419573ccadea8a/get_cpus.sh" /usr/local/nginx/
```
Comment (NickM-27) [inline docker/main/Dockerfile:34]: I think we would prefer just have the script directly inside the run file to reduce external dependencies and this docker build change. Would be good to leave the commented link so it is clear where it came from

Comment (felipecrs) [reply docker/main/Dockerfile:34]: Hm... are you sure? I'll be a little ugly. 😅

Comment (NickM-27) [reply docker/main/Dockerfile]: yes 👍

**Comment by felipecrs** [comment]
@NickM-27 btw I just noticed this:

```
# we need to catch any errors because sed will fail if user has bind mounted a custom nginx file
```

How about we avoid calling `set_worker_processes` if the nginx file is mounted? I can do the changes.

**Comment by NickM-27** [comment]
I think it might be preferable to leave it as is. Changes were made so users shouldn't need to bind mount this file anymore, but in any case probably best to catch any error that occurs so it doesn't stop frigate from starting.

**Review by NickM-27 — APPROVED** [review]

**Comment by felipecrs** [comment]
It would be very unexpected to have errors happening at that command though. I have these changes ready:

```bash
function set_worker_processes() {
    if mountpoint -q /usr/local/nginx/conf/nginx.conf; then
        echo "[INFO] NGINX configuration is mounted. Skipping worker_processes calculation..." >&2
        return
    fi

    # Capture number of assigned CPUs to calculate worker processes
    local cpus
    cpus=$(get_cpus)
    if [[ "${cpus}" -gt 4 ]]; then
        cpus=4
    fi

    echo "[INFO] Setting NGINX worker_processes to ${cpus}..."
    sed -i "s/worker_processes auto;/worker_processes ${cpus};/" /usr/local/nginx/conf/nginx.conf
}
```

Let me know in case you change your mind.

**Pull request merged**